Design_Doc_Examples/RAG_Q&A_for collaborative_work_platform.md

```diff
@@ -101,20 +101,106 @@ Optional idea to evaluate how good system is at not-clear questions with request
+
+1. For Markdown documents
+    - Expected Document Size: Up to 500 pages.
+    - Structure: Documents larger than 10 pages typically include a table of contents and dedicated sections, such as introduction or glossary.
+    - Content: Documents may include text with all Markdown features (e.g., quotes, headings, formulas, tables).
+2. For documents in Image form (scans/images): no additional description available. They are just files that contain image/scan inside.
+3. Each document have origination metadata
+4. Documents may have v1-v2-v3-... versions. For some Documents we know the diff between vX and vY. For some Documents we do not know the diff, and only know how Document looked like at each version.
+
+Clients can edit documents online via the platform or upload documents from their local machines. Each document receives a version number upon:
+
+- Saving
+- Uploading a new document
+- Uploading a version of the existing document
+
+Clients can access all versions of each document.
+
+#### ii. Data cleaning
+
+Data cleaning process should be automatized with reproducible scripts. They need to run with some schedule, regularly + on huge new document upload.
+Question: is version control needed? We already have one in the system. How to deal with that?
+
+**1. Markdown Documents:**
+- Links: do we need to save links? If yes, check that links within the document are not broken. Update or remove broken links.
+- Duplicate removal. Need to ensure that different versions of one document are not treated as duplicates despite being very similar.
+- Table of Contents (ToC): Generate or validate the presence of a ToC for documents longer than 10 pages. Tools like Pandoc or custom scripts can automate this.
+- Extract and clean the text from Markdown files, removing any Markdown syntax.
+
+**2. Scanned/Image Documents:**
+- Enhance the quality of scans (e.g., adjusting brightness/contrast, removing noise).
+- Optical Character Recognition (OCR) for scans. Is it necessary? If yes, convert scans into text with OCR (Tesseract).
+- Duplicate removal
+
+Cleaned documents and images should be stored separately from the original files in a `cleaned_data` directory. This ensures keeping the original versions for reference and debugging.
+
+#### iii. Metadata
+
+**Document Metadata:**
+- Document title
+- Author
+- Creation date
+- Last modified date
+- Table of Contents (for text documents)
+- Summary (need to discuss whether it's necessary)
+- Version history
+   - Version number
+   - Editor
+   - Version creation date
+   - Changes made in the version (if available)
+   - Diff information (if available)
+
+**Handling Metadata:**
+
+For Markdown documents, embed metadata in a YAML format at the top of each document. For images, metadata can be stored in a separate JSON file with the same name as the image.
+
+Do we need to add cleaning scripts information into the yaml?
```

Comment:
Not sure, as LLM may consume Markdown as is

Reply:
Ok, won't add it